src/app/3d/qgspointcloud3dsymbolwidget.cpp

```diff
@@ -19,29 +19,143 @@
 #include "qgspointcloud3dsymbol.h"
 #include "qgspointcloudlayer3drenderer.h"
 
-QgsPointCloud3DSymbolWidget::QgsPointCloud3DSymbolWidget( QgsPointCloud3DSymbol *symbol, QWidget *parent )
+QgsPointCloud3DSymbolWidget::QgsPointCloud3DSymbolWidget( QgsPointCloudLayer *layer, QgsPointCloud3DSymbol *symbol, QWidget *parent )
   : QWidget( parent )
+  , mLayer( layer )
 {
   setupUi( this );
 
   mPointSizeSpinBox->setClearValue( 2.0 );
 
+  for ( QgsPointCloudAttribute attr : layer->attributes().attributes() )
```

Comment:
Not needed -- use QgsPointCloudAttributeComboBox instead and its all taken care of for you (you can even apply filters to only show numeric attributes)